nixtla/nixtla_client.py

```diff
@@ -658,6 +667,20 @@ def _maybe_assign_weights(
         else:
             self.weights_x = type(df)({"features": x_cols, "weights": weights})
 
+    def _maybe_assign_feature_contributions(
+        self,
+        feature_contributions: Optional[list[list[float]]],
+        x_cols: List[str],
+        out_df: pd.DataFrame,
+    ) -> None:
+        if feature_contributions is None:
+            return
+        else:
+            shap_df = pd.DataFrame(
+                np.array(feature_contributions).T, columns=x_cols + ["base_value"]
+            )
```

Comment:
These are already column-oriented, so we can just create a dict with the column names
```suggestion
            shap_df = type(out_df)(
                dict(zip(x_cols + ["base_value"], feature_contributions))
            )
```